subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultPropertySpecFactory.java

```diff
@@ -49,7 +49,7 @@ public DeclaredTaskInputFileProperty createInputDirSpec(ValidatingValue dirPath)
     }
 
     private DeclaredTaskInputFileProperty createInputFilesSpec(ValidatingValue paths, ValidationAction validationAction) {
-        return new DefaultTaskInputFilePropertySpec(task.getName(), resolver, paths, validationAction);
+        return new DefaultTaskInputFilePropertySpec(task.toString(), resolver, paths, validationAction);
```

Comment:
If a plugin author overrides the `toString` method on their task, won't this break??